pkg/appgw/requestroutingrules.go

```diff
@@ -133,18 +135,33 @@ func (builder *appGwConfigBuilder) RequestRoutingRules(ingressList [](*v1beta1.I
 				httpsAvailable = true
 			}
 
-			// TODO check annotations for disabling http
 			if httpAvailable {
 				if wildcardRule != nil && len(rule.Host) != 0 {
 					// only add wildcard rules when host is specified
 					urlPathMaps[listenerHTTPID] = builder.pathMaps(ingress, wildcardRule,
 						listenerHTTPID, urlPathMaps[listenerHTTPID],
 						defaultAddressPoolID, defaultHTTPSettingsID)
 				}
+
 				// need to eliminate non-unique paths
 				urlPathMaps[listenerHTTPID] = builder.pathMaps(ingress, &rule,
 					listenerHTTPID, urlPathMaps[listenerHTTPID],
 					defaultAddressPoolID, defaultHTTPSettingsID)
+
+				httpURLPathMap := urlPathMaps[listenerHTTPID]
+
+				// if ssl-redirect annotation is present setup the correct redirect configuration.
+				if annotations.IsSslRedirect(ingress) {
+					sslRedirectConfigID := builder.appGwIdentifier.redirectConfigurationID(generateSSLRedirectConfigurationName(ingress.Namespace, ingress.Name))
```

Comment:
I would have carved this block out into a separate function to make it easier to test.